packages/app-wallet-interface/source/index.html.md

```diff
@@ -56,29 +56,75 @@ like `https://myserver.com/state_channel_callback`.
 Format of message sent from the wallet to the app, so that the app can then relay it
 to another participant.
 
-```javascript
-const participantId = 'abc123'; // allocated by app
-const data = '0x456';
-const message = new Message(participantId, data);
-```
-
 ```json
 {
-  "participantId": "abc123",
+  "recipient": "user123",
+  "sender": "user456",
   "data": "0x456"
 }
 ```
 
-| Parameter     | Type   | Description                                                  |
-| ------------- | ------ | ------------------------------------------------------------ |
-| participantId | String | Identifier or user that the message should be relayed to     |
-| data          | String | Message payload. Format defined by wallet and opaque to app. |
+| Parameter | Type   | Description                                                  |
+| --------- | ------ | ------------------------------------------------------------ |
+| recipient | String | Identifier of user that the message should be relayed to     |
+| sender    | String | Identifier of user that the message is from                  |
+| data      | String | Message payload. Format defined by wallet and opaque to app. |
 
 # Initial API
 
 Still a WIP! Adding methods as we need them.
 
-## List identities
+## Push Message
+
+Used to push messages received from other participants into the wallet.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "PushMessage",
+  "id": 1,
+  "params": {
+    "recipient": "user123",
+    "sender": "user456",
+    "data": "0x123.."
+  }
+}
+```
+
+> Example response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "acknowledged"
```

Comment:
While JSON-PRC allows for any type as a result, I recommend we stick to objects or arrays to keep the types more consistent in TypeScript.

```suggestion
  "result": { "success": true }
```